### PR TITLE
[NSA-9530] Add unit tests to getAudit.js

### PR DESCRIPTION
**Pull request by alexplattdfe**
- Add unit tests to cover each specific call to the getUserDetailsById/ getCachedUserById function  in getAudit, ensuring the full req object is passed in.

### CODE REVIEW

File: test/app/users/getAudit.test.js [hunk 1]
```diff
@@ -397,4 +414,167 @@ describe("when getting users audit details", () => {
...
+
+    const differentUserCall = getUserDetailsById.mock.calls.find(
+      (call) => call[0] === "DIFFERENT-USER",
+    );
+    expect(differentUserCall).toBeDefined();
+    expect(differentUserCall[1]).toBe(req);
+    expect(differentUserCall[1]).toHaveProperty("id");
+    expect(differentUserCall[1]).toHaveProperty("params");
+  });
 });
```
Comment (AMcCartneyMTH) [inline test/app/users/getAudit.test.js:786]: Please add tests to cover errors being thrown by calls to dependent functions, database calls and cache look-ups. Please consider adding another describe('... function for the tests before the end of the file.

Some tests to look at (functions **getAudit(...)** itself calls):

- **`getUserDetailsById` throws** - Handles when user fetching fails
- **`getPageOfUserAudits` throws** - Handles when audit data retrieval fails
- **`getUserOrganisationsWithServicesRaw` throws** - Handles when user organisations fetch fails
- **`getUserStatusRaw` throws** - Handles when user status retrieval fails (for deactivated users)
- **`getServiceIdForClientId` throws** - Handles when service mapping fails
- **`getServiceRaw` throws** - Handles when service detail fetching fails
- **`getOrganisationLegacyRaw` throws** - Handles when organisation lookup fails
- **`getCachedUserById` throws for edited user** - Handles errors when fetching edited user in audit events

File: test/app/users/getAudit.test.js [hunk 2]
```diff
@@ -397,4 +414,167 @@ describe("when getting users audit details", () => {
...
+    await getAudit(req, res);
+
+    const editedUserCall = getUserDetailsById.mock.calls.find(
+      (call) => call[0] === "edited-user1",
+    );
+    expect(editedUserCall).toBeDefined();
+    expect(editedUserCall[1]).toBe(req);
+    expect(editedUserCall[1]).toHaveProperty("id");
+    expect(editedUserCall[1]).toHaveProperty("params");
+  });
```
Comment (AMcCartneyMTH) [inline test/app/users/getAudit.test.js:529]: Please add a test in here to cover the edge case where an out of bound page is requested i.e. negative pages

**Review by AMcCartneyMTH — CHANGES_REQUESTED** [review]
Just a couple of points here. Whilst there is some reptation in the test set-ups, I find it helpful here because it makes things more explicit. Although I'm happy if you wanted to extract some of the setup out into a test fixture helper.

The main thing in this coverage was some missing tests around error handling.

File: test/app/users/getAudit.test.js (hunk 1 above)
Comment (AMcCartneyMTH) [reply test/app/users/getAudit.test.js:786]: The error handling tests are comprehensive and check that each dependency's errors properly propagate when exceptions are thrown. The new describe block is well-organised and makes the error test cases easy to find and maintain.

Comment (AMcCartneyMTH) [reply test/app/users/getAudit.test.js:786]: Bonus tests:
- Error handling in describeAuditEvent when processing complex audit types
- Error handling for cache operation failures during user lookups

File: test/app/users/getAudit.test.js (hunk 2 above)
Comment (AMcCartneyMTH) [reply test/app/users/getAudit.test.js:529]: This test verifies that when a negative page number ("-1") is requested, the function returns a 400 status code. Combined with the existing test at line 351 that checks for non-numeric page values, we have good coverage of invalid page input scenarios.

**Review by AMcCartneyMTH — APPROVED** [review]
We have good test coverage over getAudit. :)

- Critical paths and core functionality (happy path)
- Logic errors, transient faults and dependant component failures (sad path)

**Review by JaredAMarsh — APPROVED** [review]
I'm happy to review this change, the functionality hasn't been modified except for a page less than 1 which shouldn't cause any impact unless a user purposefully changes the URL. Additionally, the unit tests cover the specific "special cases" in terms of audit log types as well as now error propagation across the system.
